### ISSUE TITLE
Poling window is larger that the DATA [error]

### ISSUE THREAD

**Issue by guerreroMisterioso** [vlfeat/matconvnet#493]
Hi, i'm having some problems when trying to configure my first network. I have coded the following:

net.addLayer('conv1', dagnn.Conv('size', [5 5 3 20], 'hasBias', true, 'stride', [2, 2], 'pad', [0 0 0 0]), {'input'}, {'conv1'},  {'conv1f'  'conv1b'});
net.addLayer('relu1', dagnn.ReLU(), {'conv1'}, {'relu1'}, {});
net.addLayer('pool1', dagnn.Pooling('method', 'max', 'poolSize', [2, 2], 'stride', [2 2], 'pad', [0 0 0 0]), {'relu1'}, {'pool1'}, {});

net.addLayer('conv2', dagnn.Conv('size', [1 1 20 10], 'hasBias', true, 'stride', [2, 2], 'pad', [0 0 0 0]), {'pool1'}, {'conv2'},  {'conv2f'  'conv2b'});
net.addLayer('conv3', dagnn.Conv('size', [3 3 10 10], 'hasBias', true, 'stride', [2, 2], 'pad', [0 0 0 0]), {'conv2'}, {'conv3'},  {'conv3f'  'conv3b'});
net.addLayer('relu2', dagnn.ReLU(), {'conv3'}, {'relu2'}, {});
net.addLayer('pool2', dagnn.Pooling('method', 'max', 'poolSize', [1, 1], 'stride', [2 2], 'pad', [0 0 0 0]), {'conv3'}, {'pool2'}, {});

net.addLayer('classifier', dagnn.Conv('size', [1 1 10 10], 'hasBias', true, 'stride', [2, 2], 'pad', [0 0 0 0]), {'pool2'}, {'classifier'},  {'conv2f'  'conv2b'});
net.addLayer('prob', dagnn.SoftMax(), {'classifier'}, {'prob'}, {});
net.addLayer('objective', dagnn.Loss('loss', 'log'), {'prob', 'label'}, {'objective'}, {});
net.addLayer('error', dagnn.Loss('loss', 'classerror'), {'prob','label'}, 'error') ;

But at execution I get the following error: 
Error using vl_nnpool
The pooling window is larger than the DATA (including padding).

I have tested with different parameters but any of them succeeded. Any idea?

Thanks in advance.

**Comment by albanie** [comment]
This error occurs when the input to the pooling layer isn't large enough. Assuming you are running the network by passing in a cell array like this:

    inputs = {'input', input, 'label', label} ;
    net.eval(inputs)

then `input` should be an `H x W x D x N` array.  If the value of either `H` or `W` is less than 3, an error will occur at the `vl_nnpool` layer.  Anything larger than this should be fine (e.g. an input of size `96 x 96 x 3 x 10` for example).

A nice way to check the inputs are correct is to use the `DagNN.print()` function which will show you, amongst other things, the dimensions of the variables at different layers of the network.

This is the output of `net.print({'input', [96 2 3 5] })` using the network configuration above:

   var|   input|    conv1|    relu1|    pool1|    conv2|   conv3|   relu2|   pool2|classifier|    prob|label|objective|  error|
------|--------|---------|---------|---------|---------|--------|--------|--------|----------|--------|-----|---------|-------|
  dims|96x2x3x5|46x0x20x5|46x0x20x5|23x0x20x5|12x1x10x5|5x0x10x5|5x0x10x5|3x1x10x5|  2x1x10x5|2x1x10x5|  n/a|  1x1x1x5|1x1x1x5|
   mem|    11KB|       0B|       0B|       0B|      2KB|      0B|      0B|    600B|      400B|    400B|  NaN|      20B|    20B|
 fanin|       0|        1|        1|        1|        1|       1|       1|       1|         1|       1|    0|        1|      1|
fanout|       1|        1|        1|        1|        1|       2|       0|       1|         1|       2|    2|        0|      0|

Those zeros describing the size of the second dimension of the variables are going to cause mischief.
